### ISSUE TITLE
Indication for upvoting posts

### ISSUE THREAD

**Issue by OmerDavidson** [SimonHalvdansson/Harmonic-HN#178]
When clicking the upvote button, the button doesn't change. After a second you get a toast saying the upvote was successful.

Issue number 1 is there is no indication that this post is now upvoted. 

Issue number 2 is when clicking the upvote button again, I expect to remove the upvote from the post. Like I am used to from a lot of other apps. But instead what happens is I get the same toast saying the upvote was successful.

Other than that the app is a joy to use. Thank you!

**Comment by flofriday** [comment]
Well, I don't think this is technically possible.

This is because we use the API to retrieve the posts which doesn't include a flag on whether you liked it already.

We _could_ of course store it just locally but it wouldn't be synced with the webpage or other clients you use.

A way to technically archive this is to not use the API but instead request the webpage and parse the HTML.
This also has it's own set of trait-ofs like, the App might break if the website gets an update (probably unlikely with HN). And maybe beeing slower.

Source: I implemented some improvements on the toasts and wanted to implement that feature too 😅

**Comment by SimonHalvdansson** [comment]
Yeah this issue is annoying and as you say @flofriday syncing is probably the biggest issue - here some background updating of liked posts or whatever seems like a lot of work for a small gain. The feature request is however totally reasonable.

**Comment by SimonHalvdansson** [comment]
Fixed now in 3.0. The app maintains an index of posts you've voted on and uses this to color in the upvote button if a upvote has been registred (syncing with the website is based on the Upvoted dropdown option, so not automatic, but still pretty good).

Removing upvotes is now active as well, so I think this is done enough to consider closeable. Maybe a periodic sync sometime in the future, but I don't love that concept.
